lib/bson/timestamp.rb

```diff
@@ -110,6 +110,8 @@ def as_extended_json(**options)
     #
     # @since 2.0.0
     def initialize(seconds, increment)
+      if seconds < 0 then seconds += 2**32 end
+      if increment < 0 then increment += 2**32 end
```

Comment:
I don't think either of these values should be negative. If they are negative in C code, they need to be fixed in C code also.

Reply:
do you think I should create a get_uint32 then?

Reply:
That sounds like a reasonable approach.